example-all.js

```diff
@@ -2,20 +2,26 @@
 var logUpdate = require('log-update');
 var cliSpinners = require('./');
 var spinners = Object.keys(cliSpinners);
-var i = 0;
-var j = 0;
-var frames = [];
+var frame = 0;
+var spinner = 0;
+var next;
 
-console.log(spinners.length + ' spinners\n');
+var showNextFrame = function () {
+	var frames = cliSpinners[spinners[spinner]].frames;
+	logUpdate(frames[frame++ % frames.length] + ' ' + spinners[spinner]);
+};
 
-setInterval(function () {
-	if (++i === frames.length) {
-		j++;
-		i = 0;
+var showNextSpinner = function () {
+	if (next) {
+		clearInterval(next);
+		spinner++;
 	}
+	if (spinner in spinners) {
```

Comment:
I like `x in a` because it is more robust and more readable. Also, AFAIK, access to array by numeric indexes get's `toString()`ed internally. And [according to MDN, it's perfectly fine to do `x in a` on `Array`s](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#Description).